### ISSUE TITLE
NN documentation

### ISSUE THREAD

**Issue by cjsmeele** [cjsmeele/hersenschemen#1]
C++ templates :unicorn::unicorn: are not self-documenting. This is mostly an issue in `neuralnet-f`.
(better write stuff down before we forget what we wrote!)

- [ ] Document neuralnet-oo
- [ ] Document neuralnet-f

**Comment by ManDeJan** [comment]
I disagree and think templates are a perfect example of expressive coding and self documenting code, it provides clear and useful errors when you use it incorrectly and thus needs no further documentation
